File: backend/internal/helpers/agent_debug_log.go
package helpers

import (
	"encoding/json"
	"os"
	"path/filepath"
	"time"
)

// #region agent log
func AgentDebugLog(location, message, hypothesisID, runID string, data map[string]any) {
	payload := map[string]any{
		"sessionId":    "0764c5",
		"location":     location,
		"message":      message,
		"hypothesisId": hypothesisID,
		"runId":        runID,
		"data":         data,
		"timestamp":    time.Now().UnixMilli(),
	}
	line, err := json.Marshal(payload)
	if err != nil {
		return
	}
	paths := []string{
		"debug-0764c5.log",
		filepath.Join("..", "debug-0764c5.log"),
	}
	for _, p := range paths {
		f, err := os.OpenFile(p, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
		if err != nil {
			continue
		}
		_, _ = f.Write(append(line, '\n'))
		_ = f.Close()
		return
	}
}

// #endregion

File: apps/mobile/src/lib/agentDebugLog.ts
const DEBUG_ENDPOINT = 'http://127.0.0.1:7774/ingest/4b4823e8-e059-45d4-bd4e-f7b6e10474eb'
const SESSION_ID = '2533bd'

/** Fire-and-forget debug log for agent sessions (NDJSON ingest on dev machine). */
export function agentDebugLog(
  location: string,
  message: string,
  data: Record<string, unknown>,
  hypothesisId: string,
  runId = 'pre-fix',
): void {
  // #region agent log
  fetch(DEBUG_ENDPOINT, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'X-Debug-Session-Id': SESSION_ID,
    },
    body: JSON.stringify({
      sessionId: SESSION_ID,
      runId,
      hypothesisId,
      location,
      message,
      data,
      timestamp: Date.now(),
    }),
  }).catch(() => {})
  // #endregion
}
